test/test_integration_ssl.rb

```diff
@@ -189,6 +188,34 @@ def test_ssl_run_with_localhost_authority
     end
   end
 
+  def test_ssl_run_without_any_key_and_authority; require 'logger'
+    skip_if :windows; require 'stringio'
```

Comment:
Why are these `require`s here? Are they needed? If so, they should not hide here at the end of these lines

Reply:
I copied this from the test `test_ssl_run_with_curl_client`.
But I will change this